src/main/groovy/de/dkfz/roddy/bam2fastq/BamToFastqWorkflow.groovy

```diff
@@ -82,13 +82,30 @@ class BamToFastqWorkflow extends Workflow {
         readGroupsPerBamfile[bamfileName]
     }
 
+    /** Update the ReadGroups with the names of the files as provided in a FileGroup. */
+    private ReadGroupGroup<BaseFile> updatedReadGroupGroup(final ReadGroupGroup<BaseFile> oldGroups, final FileGroup fileGroup) {
```

Comment:
Why did you move this away from the readgroup? I don't think it should be located in the workflow. Then maybe use a helper class or so.

Reply:
It does not belong into ReadGroupGroup either, as this is generic with unconstrained T but this code is specific for T extends BaseFile. I was thinking of a subclass, but that did not seem to be much better. The problem is also the FileGroup class, that has nothing to do with the ReadGroups. In this sense, this code does actually not belong to ReadGroupGroup.

Reply:
Marked as deprecated.